tools/cldr-apps/src/main/java/org/unicode/cldr/web/SurveyMain.java

```diff
@@ -2314,41 +2316,43 @@ private void doMain(WebContext ctx) {
 
     private static ExampleGenerator gComparisonValuesExample = null;
 
-    private Factory gFactory = null;
+    private Supplier<Factory> gSupplementalDiskFactory =
+            Suppliers.memoize(() -> SimpleFactory.make(getCommonMainFileBase(), ".*"));
 
-    /**
-     * @return a Factory (not a singleton) just for use with supplemental data
-     */
-    public Factory getSupplementalDiskFactory() {
-        return SimpleFactory.make(getCommonMainFileBase(), ".*");
+    /** Factory for use with disk data. Singleton */
+    public final Factory getSupplementalDiskFactory() {
+        return gSupplementalDiskFactory.get();
     }
 
     /**
      * Return the factory that corresponds to trunk
      *
      * @return
      */
-    public synchronized Factory getDiskFactory() {
-        if (gFactory == null) {
-            final File[] list = getFileBases();
-            CLDRConfig config = CLDRConfig.getInstance();
-            // may fail at server startup time- should do this through setup mode
-            ensureOrCheckout(config.getCldrBaseDirectory());
-            // verify readable
-            File root = new File(config.getCldrBaseDirectory(), "common/main");
-            if (!root.isDirectory()) {
-                throw new InternalError(
-                        "Not a dir:  "
-                                + root.getAbsolutePath()
-                                + " - check the value of "
-                                + CldrUtility.DIR_KEY
-                                + " in cldr.properties.");
-            }
+    public final Factory getDiskFactory() {
+        return gFactory.get();
+    }
+
+    private Supplier<Factory> gFactory =
+            Suppliers.memoize(
+                    () -> {
+                        final File[] list = getFileBases();
+                        CLDRConfig config = CLDRConfig.getInstance();
+                        // may fail at server startup time- should do this through setup mode
```

Comment:
"may fail at server startup time" -- if so, what will happen?

Reply:
It'll actually explain to the user that the survey tool isn't configured yet and invite hem to Edit cldr.properties.

Reply:
so it would only be expected to fail if cldr.properties isn't properly configured yet? if so, the comment could be less alarming if it spelled that out...